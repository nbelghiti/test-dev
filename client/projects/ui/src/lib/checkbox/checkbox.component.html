<label
	class="c-checkbox"
	[ngClass]="{
		'is-indeterminate': indeterminate
	}"
>
	<input
		[id]="name"
		[name]="name"
		class="c-checkbox__input"
		type="checkbox"
		[formControl]="isChecked"
	/>

	<span class="c-checkbox__box">
		<span class="c-checkbox__check">
			<ui-icon [name]="indeterminate ? IconName.minus : IconName.check" [size]="IconSize.tiny"></ui-icon>
		</span>
	</span>

	<span *ngIf="label" class="c-checkbox__label">{{ label }}</span>
</label>
