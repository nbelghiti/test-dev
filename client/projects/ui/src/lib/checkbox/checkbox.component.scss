// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================
$checkbox-size: 1.7rem;

// =============================================================================
// :: Checkbox
// =============================================================================
.c-checkbox {
	display: inline-flex;
	align-items: center;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-checkbox__input {
	position: absolute;

	width: 0;
	height: 0;

	opacity: 0;

	&:checked {

		+ .c-checkbox__box {
			background-color: $primary-color;
			border-color: $primary-color;

			.c-checkbox__check {
				opacity: 1;

				transform: scale(1);
			}
		}
	}
}

.c-checkbox__box {
	display: inline-flex;
	justify-content: center;
	align-items: center;
	width: $checkbox-size;
	height: $checkbox-size;

	border-radius: 0.3rem;

	border: 0.2rem solid $border-color-grey-76;

	transition:
		border-color $animation-base,
		background-color $animation-base;

	cursor: pointer;

	&:hover {
		border-color: $border-color-black;
	}
}

.c-checkbox__check {
	display: flex;

	line-height: $checkbox-size;

	color: $background-color-base;
	opacity: 0;

	transform: scale(0);

	transition:
		opacity $animation-base,
		transform $animation-base;
}

.c-checkbox__label {
	margin-left: 1.2rem;

	white-space: nowrap;
	font-size: $font-size-base;
	line-height: $line-height-normalize;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-checkbox.is-indeterminate {

	.c-checkbox__input {

		+ .c-checkbox__box {
			opacity: 0.5;
		}
	}
}
