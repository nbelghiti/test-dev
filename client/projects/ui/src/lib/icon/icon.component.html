<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
