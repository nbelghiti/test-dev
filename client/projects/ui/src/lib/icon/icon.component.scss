// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
