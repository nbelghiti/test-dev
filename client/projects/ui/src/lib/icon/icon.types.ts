export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
