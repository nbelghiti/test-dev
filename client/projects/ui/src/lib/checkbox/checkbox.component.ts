import { Component, Input, OnDestroy, forwardRef } from '@angular/core';
import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

import { IconName, IconSize } from '../icon/icon.types';

@Component({
	selector: 'ui-checkbox',
	templateUrl: './checkbox.component.html',
	styleUrls: ['./checkbox.component.scss'],
	providers: [
		{
			provide: NG_VALUE_ACCESSOR,
			useExisting: forwardRef(() => CheckboxComponent),
			multi: true,
		},
	],
})

export class CheckboxComponent implements OnDestroy, ControlValueAccessor {
	@Input() public indeterminate: boolean = false;
	@Input() public label: string;
	@Input() public name: string = 'checkbox-input';

	public IconName = IconName;
	public IconSize = IconSize;

	public isChecked: FormControl = new FormControl(false);

	private destroyed$: Subject<boolean> = new Subject<boolean>();

	constructor() {
		this.isChecked.valueChanges
			.pipe(takeUntil(this.destroyed$))
			.subscribe((isChecked: boolean) => this.onChange(isChecked));
	}

	public onChange: (_) => any = () => { };

	public onTouched: (_) => any = () => { };

	public writeValue(isChecked: boolean): void {
		if (this.isChecked.value !== isChecked) {
			this.isChecked.setValue(isChecked, { emitEvent: false });
		}
	}

	public registerOnChange(onChange: (_) => any): void {
		this.onChange = onChange;
	}

	public registerOnTouched(onTouched: () => void): void {
		this.onTouched = onTouched;
	}

	public ngOnDestroy(): void {
		this.destroyed$.next(true);
		this.destroyed$.complete();
	}
}
