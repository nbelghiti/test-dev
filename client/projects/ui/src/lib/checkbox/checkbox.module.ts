import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';

import { IconModule } from '../icon/icon.module';

import { CheckboxComponent } from './checkbox.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		ReactiveFormsModule,
	],
	declarations: [
		CheckboxComponent,
	],
	exports: [
		CheckboxComponent,
	],
})
export class CheckboxModule {}
