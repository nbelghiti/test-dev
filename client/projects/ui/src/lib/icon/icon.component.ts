import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
