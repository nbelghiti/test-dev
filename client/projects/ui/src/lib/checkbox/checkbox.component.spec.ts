import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { IconComponent } from '../icon/icon.component';

import { CheckboxComponent } from './checkbox.component';

describe('UserInfoComponent', () => {
	let component: CheckboxComponent;
	let fixture: ComponentFixture<CheckboxComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [ ReactiveFormsModule ],
			declarations: [
				CheckboxComponent,
				IconComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(CheckboxComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

