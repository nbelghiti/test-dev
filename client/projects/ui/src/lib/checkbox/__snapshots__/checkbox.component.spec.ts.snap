// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoComponent Should render correctly 1`] = `
<ui-checkbox
  IconName={[Function Object]}
  IconSize={[Function Object]}
  destroyed$={[Function Subject]}
  indeterminate="false"
  isChecked={[Function FormControl]}
  name={[Function String]}
  onChange={[Function Function]}
  onTouched={[Function Function]}
>
  <label
    class="c-checkbox"
    ng-reflect-klass="c-checkbox"
    ng-reflect-ng-class="[object Object]"
  >
    <input
      class="c-checkbox__input ng-untouched ng-pristine ng-valid"
      id="checkbox-input"
      name="checkbox-input"
      ng-reflect-form="[object Object]"
      type="checkbox"
    />
    <span
      class="c-checkbox__box"
    >
      <span
        class="c-checkbox__check"
      >
        <ui-icon
          class="c-icon"
          ng-reflect-name="check"
          ng-reflect-size="tiny"
        >
          <i
            class="c-icon__glyph c-icon__glyph--tiny  fas fa-check"
          />
        </ui-icon>
      </span>
    </span>
    
  </label>
</ui-checkbox>
`;
